cl/corpus_importer/management/commands/columbia_merge.py

```diff
@@ -0,0 +1,865 @@
+    """
+
+    if not overlapping_data:
+        return {}
+
+    columbia_data, cl_data = overlapping_data
+    # We check if any word in the string is uppercase
+    cl_data_upper = (
+        True if [s for s in cl_data.split(",") if s.isupper()] else False
+    )
+
+    # Get last names keeping case and cleaning the string (We could have
+    # the judge names in capital letters)
+    cl_clean = set(find_all_judges(cl_data))
+    # Lowercase courtlistener judge names for set operations
+    temp_cl_clean = set([c.lower() for c in cl_clean])
+    # Get last names in lowercase and cleaned
+    columbia_clean = set(find_all_judges(columbia_data))
+    # Lowercase columbia judge names for set operations
+    temp_columbia_clean = set([h.lower() for h in columbia_clean])
+    # Prepare judges string
+    judges = titlecase(", ".join(find_all_judges(columbia_data)))
+
+    if (
+        temp_columbia_clean.issuperset(temp_cl_clean) or cl_data_upper
+    ) and columbia_clean != cl_clean:
+        return {"judges": judges}
+    elif not temp_columbia_clean.intersection(temp_cl_clean):
+        # Last resort, use distance between words to solve typos
+        cl_clean_list = list(cl_clean)
+        columbia_clean_list = list(columbia_clean)
+        judge_pairs = list(
+            itertools.product(cl_clean_list, columbia_clean_list)
+        )
+        success = False
+        for pair in judge_pairs:
+            s = SequenceMatcher(None, pair[0].lower(), pair[1].lower())
+            if s.ratio() >= 0.7:
+                # We found a match, we assume that the data in columbia is better,
+                # we keep the one from columbia and remove the one from cl
+                try:
+                    cl_clean_list.remove(pair[0])
+                except ValueError:
+                    # The value was removed in an earlier iteration, but we still
+                    # have the value in the remaining pairs to match, we simply
+                    # ignore it
+                    pass
+                success = True
+
+        if success:
+            # At least one success that matches the names, we can create a new judges
+            # list
+            new_judges_list = sorted(
+                list(set(cl_clean_list + columbia_clean_list))
+            )
+            return {"judges": titlecase(", ".join(new_judges_list))}
+        else:
+            raise JudgeException("Judges are completely different.")
+
+    return {}
+
+
+def merge_overlapping_data(
+    cluster: OpinionCluster, changed_values_dictionary: dict
+) -> dict[str, Any]:
+    """Merge overlapping data
+    :param cluster: the cluster object
+    :param changed_values_dictionary: the dictionary of data to merge
+    """
+
+    if not changed_values_dictionary:
+        # Empty dictionary means that we don't have overlapping data
+        return {}
+
+    long_fields = ["syllabus", "posture"]
+
+    data_to_update = {}
+
```

Comment:
same here... is this harvard code - duplicated?  If so let's move it to a util file or something so we dont duplicate our work.

Reply:
or call that code.

Reply:
yes, i moved to corpus_importer/utils.py so that it can be reused with harvard/columbia